packages/2d/src/lib/components/Icon.ts

```diff
@@ -54,8 +54,17 @@ export class Icon extends Img {
   @colorSignal()
   public declare color: ColorSignal<this>;
 
+  /**
+   * The init check is only used to supress an error warning about setting the `src` Property.
+   */
+  private init = false;
+
   public constructor(props: IconProps) {
-    super(props);
+    super({
+      ...props,
+      src: null!,
```

Comment:
Anytime I see `!` bypassing the type system, I wonder if there's a different way. Maybe we check src where it's used and emit the error there, instead of making initialization / error checking more complex (and bypassing type checking at the same time).

Maybe the ideal is we set up an overridable function in `Node` called checkProps that is called after props are set and can be overridden to add or remove the error message about `src`.

Reply:
In this specific case it might make sense to allow `src` to accept `null`, as the warning message indicates that one can do so:

![image](https://github.com/motion-canvas/motion-canvas/assets/27686785/4903c244-4027-451b-8eb3-db7c3c797ac6)

Reply:
I think that would be a fine change as well, but should be one that updates the allowed types as well instead of bypassing them 🙂